### PR TITLE
Improve next_pass documentation

### PR DESCRIPTION
**Pull request by huwpascoe**

### CODE REVIEW

File: tutorials/3d/standard_material_3d.rst [hunk 1]
```diff
@@ -734,12 +734,17 @@ Material Settings
 Render priority
 ---------------
 
-The rendering order of objects can be changed, although this is mostly
+This is mostly
```
Comment (AThousandShips) [inline tutorials/3d/standard_material_3d.rst:737]: Why this change? In either case this new sentence is not very clear, the "this" part is confusing IMO when nothing is referred to

File: tutorials/3d/standard_material_3d.rst [hunk 2]
```diff
@@ -734,12 +734,17 @@ Material Settings
...
 useful for transparent objects (or opaque objects that perform depth draw
 but no color draw, such as cracks on the floor).
 
+Objects are Sorted by opaque/transparent queue, then render_priority. With higher priority being drawn later. Depth testing overrules priority. Priority alone cannot force opaque objects to be drawn over eachother.
```
Comment (AThousandShips) [inline tutorials/3d/standard_material_3d.rst:741]:
```suggestion
Objects are Sorted by opaque/transparent queue, then :ref:`render_priority<class_Material_property_render_priority>`. With higher priority being drawn later. Depth testing overrules priority. Priority alone cannot force opaque objects to be drawn over eachother.
```
Or
```suggestion
Objects are Sorted by opaque/transparent queue, then ``render_priority``. With higher priority being drawn later. Depth testing overrules priority. Priority alone cannot force opaque objects to be drawn over eachother.
```

File: tutorials/3d/standard_material_3d.rst [hunk 3]
```diff
@@ -734,12 +734,17 @@ Material Settings
...
+
 Next Pass
 ---------
 
-Sets the material to be used for the next pass. This renders the object
-again with a different material.
+Setting next_pass material will cause an object to be rendered again with that next material. Sorted by opaque/transparent queue, then render_priority, then depth.
```
Comment (AThousandShips) [inline tutorials/3d/standard_material_3d.rst:746]: Same here

File: tutorials/3d/standard_material_3d.rst [hunk 4]
```diff
@@ -734,12 +734,17 @@ Material Settings
...
+
+.. image:: img/next_pass.webp
+
+Depth will test equal between both materials unless the grow setting or other vertex transformations are used. Multiple transparent passes should use render_priority to ensure correct ordering.
```
Comment (AThousandShips) [inline tutorials/3d/standard_material_3d.rst:750]: And here

File: tutorials/3d/standard_material_3d.rst (hunk 2 above)
Comment (AThousandShips) [reply tutorials/3d/standard_material_3d.rst:741]: Also this line needs to be broken up, it's too long

File: tutorials/3d/standard_material_3d.rst (hunk 1 above)
Comment (huwpascoe) [reply tutorials/3d/standard_material_3d.rst:737]: Ah. It was a run on sentence, originally I'd put the other one first. Will fix.

File: tutorials/3d/standard_material_3d.rst [hunk 5]
```diff
@@ -738,8 +738,19 @@ The rendering order of objects can be changed, although this is mostly
 useful for transparent objects (or opaque objects that perform depth draw
 but no color draw, such as cracks on the floor).
 
+Objects are Sorted by opaque/transparent queue, then :ref:`render_priority<class_Material_property_render_priority>`.
+With higher priority being drawn later.
+
+Depth testing overrules priority. Priority alone cannot force opaque objects to be drawn over eachother.
```
Comment (AThousandShips) [inline tutorials/3d/standard_material_3d.rst:744]:
```suggestion
Depth testing overrules priority. Priority alone cannot force opaque objects to be drawn over each other.
```

File: tutorials/3d/standard_material_3d.rst [hunk 6]
```diff
@@ -738,8 +738,19 @@ The rendering order of objects can be changed, although this is mostly
 useful for transparent objects (or opaque objects that perform depth draw
 but no color draw, such as cracks on the floor).
 
+Objects are Sorted by opaque/transparent queue, then :ref:`render_priority<class_Material_property_render_priority>`.
```
Comment (clayjohn) [inline tutorials/3d/standard_material_3d.rst:741]:
```suggestion
Objects are sorted by opaque/transparent queue, then :ref:`render_priority<class_Material_property_render_priority>`.
```

File: tutorials/3d/standard_material_3d.rst [hunk 7]
```diff
@@ -738,8 +738,19 @@ The rendering order of objects can be changed, although this is mostly
...
+With higher priority being drawn later.
+
+Depth testing overrules priority. Priority alone cannot force opaque objects to be drawn over each other.
+
 Next Pass
 ---------
 
-Sets the material to be used for the next pass. This renders the object
-again with a different material.
+Setting :ref:`next_pass<class_Material_property_next_pass>` material will cause an object to be rendered again with that next material.
```
Comment (clayjohn) [inline tutorials/3d/standard_material_3d.rst:749]:
```suggestion
Setting :ref:`next_pass<class_Material_property_next_pass>` on a material will cause an object to be rendered again with that next material.
```

File: tutorials/3d/standard_material_3d.rst [hunk 8]
```diff
@@ -738,8 +738,19 @@ The rendering order of objects can be changed, although this is mostly
...
+
+Sorted by opaque/transparent queue, then :ref:`render_priority<class_Material_property_render_priority>`, then depth.
```
Comment (clayjohn) [inline tutorials/3d/standard_material_3d.rst:751]:
```suggestion
Objects are sorted by opaque/transparent queue, then :ref:`render_priority<class_Material_property_render_priority>`. Transparent objects are also sorted by depth.
```

**Review by clayjohn — COMMENTED** [review]
Just some spelling/grammar suggestions. Otherwise it looks good

**Comment by huwpascoe** [comment]
All the grammatical errors are fixed I think.

File: tutorials/3d/standard_material_3d.rst [hunk 9]
```diff
@@ -738,8 +738,21 @@ The rendering order of objects can be changed, although this is mostly
 useful for transparent objects (or opaque objects that perform depth draw
 but no color draw, such as cracks on the floor).
 
+Objects are sorted by opaque/transparent queue, then :ref:`render_priority<class_Material_property_render_priority>`.
```
Comment (Calinou) [inline tutorials/3d/standard_material_3d.rst:741]:
```suggestion
Objects are sorted by an opaque/transparent queue, then :ref:`render_priority<class_Material_property_render_priority>`,
```

File: tutorials/3d/standard_material_3d.rst [hunk 10]
```diff
@@ -738,8 +738,21 @@ The rendering order of objects can be changed, although this is mostly
...
+With higher priority being drawn later.
```
Comment (Calinou) [inline tutorials/3d/standard_material_3d.rst:742]:
```suggestion
with higher priority being drawn later.
```

**Review by Calinou — COMMENTED** [review]
Should be good to merge after applying suggestions.

**Comment by Calinou** [comment]
Do you have the project files to generate that image? If so, could you upload it here as a ZIP for future reference?

**Comment by huwpascoe** [comment]
Here is source [next_pass.zip](https://github.com/user-attachments/files/16736385/next_pass.zip) (inkscape file)

**Review by Calinou — APPROVED** [review]
Thanks!

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.3 in https://github.com/godotengine/godot-docs/pull/10038.
